File: experiments_dataset_vul_detect_cwe_classification/dataset/vul_dataset/java/non-vul/code_before11.java
// From cwe-snippets, snippets_90/non-compliant/Java/cwe-0459/temp_file_01.java

private void func() throws Throwable
{

    File tempFile = null;

    try
    {
        tempFile = File.createTempFile("temp", "1234");
        IO.writeLine(tempFile.toString());

        /* FIX: Call deleteOnExit() so that the file will be deleted */
        tempFile.deleteOnExit();

        /* Set the permissions to avoid insecure temporary file incidentals  */
        if (!tempFile.setWritable(true, true))
        {
            IO.logger.log(Level.WARNING, "Could not set Writable permissions");
        }
        if (!tempFile.setReadable(true, true))
        {
            IO.logger.log(Level.WARNING, "Could not set Readable permissions");
        }
        if (!tempFile.setExecutable(false))
        {
            IO.logger.log(Level.WARNING, "Could not set Executable permissions");
        }
    }
    catch (IOException exceptIO)
    {
        IO.logger.log(Level.WARNING, "Could not create temporary file", exceptIO);
    }

}

File: experiments_dataset_vul_detect_cwe_classification/dataset/vul_dataset/c/non-vul/code_before11.c
// From cwe-snippets, snippets_90/non-compliant/C/cwe-0476/binary_if_03.c

static void func()
{
    if(5==5)
    {
        {
            twoIntsStruct *twoIntsStructPointer = NULL;
            if ((twoIntsStructPointer != NULL) && (twoIntsStructPointer->intOne == 5))
            {
                printLine("intOne == 5");
            }
        }
    }
}